From cd9f56675bd1c394c1a07bf1c760ccef485d10cb Mon Sep 17 00:00:00 2001
From: Salvo Giangreco <giangrecosalvo9@gmail.com>
Date: Mon, 17 Nov 2025 00:46:42 +0100
Subject: [PATCH] Enable quick switch to Secure Folder feature

One UI 9 already adds SecureFolderAction unconditionally in
FunctionKeyUtils. Only its Settings search metadata remains gated by the
presence of aliveprivacy.

---
 .../android/settings/usefulfeature/Usefulfeature$1.smali | 9 ---------
 1 file changed, 9 deletions(-)

diff --git a/smali_classes3/com/samsung/android/settings/usefulfeature/Usefulfeature$1.smali b/smali_classes3/com/samsung/android/settings/usefulfeature/Usefulfeature$1.smali
--- a/smali_classes3/com/samsung/android/settings/usefulfeature/Usefulfeature$1.smali
+++ b/smali_classes3/com/samsung/android/settings/usefulfeature/Usefulfeature$1.smali
@@ -251,14 +251,6 @@
 
     iput-object v1, v0, Lcom/android/settingslib/search/SearchIndexableRaw;->keywords:Ljava/lang/String;
 
-    const-string v1, "com.samsung.android.aliveprivacy"
-
-    invoke-static {p1, v1}, Lcom/android/settings/Utils;->hasPackage(Landroid/content/Context;Ljava/lang/String;)Z
-
-    move-result v1
-
-    if-eqz v1, :cond_1
-
     new-instance v1, Ljava/lang/StringBuilder;
 
     invoke-direct {v1}, Ljava/lang/StringBuilder;-><init>()V
@@ -281,7 +273,6 @@
 
     iput-object p1, v0, Lcom/android/settingslib/search/SearchIndexableRaw;->keywords:Ljava/lang/String;
 
-    :cond_1
     const-string p1, "function_key_setting"
 
     iput-object p1, v0, Landroid/provider/SearchIndexableData;->key:Ljava/lang/String;
-- 
2.51.2
